### PR TITLE
rhsm-conduit-cron job Openshift def is converted to use rest endpoint

### PR DESCRIPTION
**Pull request by chaynesred**
https://issues.redhat.com/browse/ENT-5030

### CODE REVIEW

File: deploy/rhsm-clowdapp.yaml [hunk 1]
```diff
@@ -1563,6 +1563,53 @@ objects:
...
+      podSpec:
+       image: ${CURL_CRON_IMAGE}:${CURL_CRON_IMAGE_TAG}
+        command:
+         - /usr/bin/bash
+           - -c
+           - curl
+           - >
+         - curl
+         - curl -X 'POST'   http://rhsm-conduit:8000/api/rhsm-subscriptions/v1internal/rpc/syncAllOrgs   -H 'accept: application/vnd.api+json'   -H 'Content-Type: application/json'   -H 'x-rh-swatch-psk: $SWATCH_SELF_PSK’
+
```
Comment (kflahert) [inline deploy/rhsm-clowdapp.yaml:1581]: Need to remove extra line here

File: deploy/rhsm-clowdapp.yaml [hunk 2]
```diff
@@ -1563,6 +1563,53 @@ objects:
...
+      restartPolicy: Never
+      podSpec:
+       image: ${CURL_CRON_IMAGE}:${CURL_CRON_IMAGE_TAG}
+        command:
+         - /usr/bin/bash
+           - -c
+           - curl
+           - >
+         - curl
+         - curl -X 'POST'   http://rhsm-conduit:8000/api/rhsm-subscriptions/v1internal/rpc/syncAllOrgs   -H 'accept: application/vnd.api+json'   -H 'Content-Type: application/json'   -H 'x-rh-swatch-psk: $SWATCH_SELF_PSK’
```
Comment (kflahert) [inline deploy/rhsm-clowdapp.yaml:1580]: A typo here: need a '/' after v1 and then I think we also need brackets around the env var like '${SWATCH_SELF_PSK}'

File: templates/rhsm-conduit.yaml [hunk 3]
```diff
@@ -331,52 +343,25 @@ objects:
...
-              - image: ${IMAGE}:${IMAGE_TAG}
+              - image: ${CURL_CRON_IMAGE}:${CURL_CRON_IMAGE_TAG}
                 name: rhsm-conduit-cron-sync
+                command:
+                  - /usr/bin/bash
+                    - -c
+                    - curl
+                    - >
+                  - curl
+                  - curl -X 'POST'   http://rhsm-conduit:8000/api/rhsm-subscriptions/v1internal/rpc/syncAllOrgs   -H 'accept: application/vnd.api+json'   -H 'Content-Type: application/json'   -H 'x-rh-swatch-psk: $SWATCH_SELF_PSK’
```
Comment (kflahert) [inline templates/rhsm-conduit.yaml:354]: Need same fixes as above in clowapp.yaml

File: deploy/rhsm-clowdapp.yaml [hunk 4]
```diff
@@ -1563,6 +1577,32 @@ objects:
...
+        activeDeadlineSeconds: 4200
+        successfulJobsHistoryLimit: 2
+        restartPolicy: Never
+        podSpec:
+          image: ${CURL_CRON_IMAGE}:${CURL_CRON_IMAGE_TAG}
+          command:
+            - /usr/bin/bash
+            - -c
+            - >
+              /usr/bin/curl --fail -H "x-rh-swatch-psk: ${SWATCH_SELF_PSK}" -X POST "http://${RHSM_SYSTEM_CONDUIT_SERVICE_HOST}:${RHSM_SYSTEM_CONDUIT_SERVICE_PORT}/api/rhsm-subscriptions/v1/internal/rpc/syncAllOrgs"
```
Comment (lindseyburnett) [inline deploy/rhsm-clowdapp.yaml:1591]: What is setting `RHSM_SYSTEM_CONDUIT_SERVICE_HOST` and `RHSM_SYSTEM_CONDUIT_SERVICE_PORT`?  I haven't deployed to EE to confirm yet, but they're looking null.

**Comment by lindseyburnett** [comment]
There are some automated test failures with `ci.ext.devshift.net PR build`.  Could you update the description of this PR with manual test steps?

**Review by lindseyburnett — CHANGES_REQUESTED** [review]
There are some automated test failures with `ci.ext.devshift.net PR build`.  Could you update the description of this PR with manual test steps?

**Comment by kflahert** [comment]
Test Steps:

- Copy to ~/.config/bonfire/config.yaml (update rhsm-subscriptions repo path)

```
appsFile:
  host: gitlab
  repo: insights-platform/cicd-common
  path: bonfire_configs/ephemeral_apps.yaml

apps:
- name: rhsm-subscriptions
  components:
    - name: capacity-allowlist
      host: gitlab
      repo: rhsm/swatch-product-allowlist
      path: templates/capacity-allowlist.yml
    - name: rhsm-subscriptions
      host: local
      repo: /home/username/IdeaProjects/rhsm-subscriptions
      path: deploy/rhsm-clowdapp.yaml
      parameters:
          REPLICAS: 1
          RHSM_RBAC_USE_STUB: 'true'
          RH_MARKETPLACE_MANUAL_SUBMISSION_ENABLED: 'true'
          DEV_MODE: 'true'
          ENABLE_ACCOUNT_RESET: 'true'
```

- replace line 1151 of rhsm-subscriptions/deploy/rhsm-clowdapp.yaml with `image: ${CONDUIT_IMAGE}:latest` (not sure why I had to do this but conduit was giving me issues)
- Set env vars
```
IMAGE=quay.io/cloudservices/rhsm-subscriptions
IMAGE_TAG=latest
BONFIRE_APP_NAME=rhsm-subscriptions
BONFIRE_COMPONENT_NAME=rhsm-subscriptions
NAMESPACE=$(bonfire namespace reserve -d 12h)
```

- Deploy to EE 
`bonfire deploy -n $NAMESPACE --no-remove-resources $BONFIRE_COMPONENT_NAME -p $BONFIRE_COMPONENT_NAME/IMAGE=$IMAGE -i $IMAGE=$IMAGE_TAG $BONFIRE_APP_NAME`

- Wait for system-conduit pod to start

- In openshift web console go to Cron Jobs -> rhsm-conduit-cron-sync -> edit Yaml and change schedule to be the next minute of the hour(ex: its 3:26 change schedule to 27 * * * *) so that the job runs in the next minute

- Check cron job pod for success message in logs

**Review by lindseyburnett — APPROVED** [review]

**Pull request merged**